### PR TITLE
Fix artifact registration in Releases API for Teleport Connect

### PR DESCRIPTION
**Pull request by justinas**
`Teleport Connect-X.Y.Z.dmg` has recently been added to CI builds. This change fixes a couple of things to support it:
* Support artifacts with spaces in the name
* Assign Teleport Connect to both `teleport` (OSS) and `teleport-ent` (Enterprise) releases - same logic as for `tsh` packages.
* Add custom artifact description for Teleport Connect (aligned with what [Houston currently has](https://github.com/gravitational/houston/blob/bddea97761a32679c9a3bec358da3f9f3698b5a6/web/src/components/releases/release.jsx#L231-L233)).

This is undoubtedly somewhat hacky, but I hope to re-code this logic in a nicer way with the move to Go (https://github.com/gravitational/cloud/issues/1504). This change only concerns master and branch/v10, but I will try to port to older branches as well for consistency (also minimizes merge conflicts down the line).

All Dronegen changes auto-generated.

### CODE REVIEW

File: dronegen/tag.go [hunk 1]
```diff
@@ -375,7 +375,7 @@ for file in $(find . -type f ! -iname '*.sha256' ! -iname '*-unsigned.zip*'); do
       cat $WORKSPACE_DIR/curl_out.txt
       exit 1
     fi
-    curl $CREDENTIALS --fail -o /dev/null -X PUT "$RELEASES_HOST/releases/$product@$VERSION/assets/$(basename $file)"
+    curl $CREDENTIALS --fail -o /dev/null -X PUT "$RELEASES_HOST/releases/$product@$VERSION/assets/$(basename "$file" | sed 's/ /%%20/g')"
```
Comment (justinas) [inline dronegen/tag.go:378]: Curl encodes data (body or url query), but not URLs, it expects them pre-encoded.

File: .drone.yml [hunk 2]
```diff
@@ -1228,7 +1231,7 @@ steps:
           cat $WORKSPACE_DIR/curl_out.txt
           exit 1
         fi
-        curl $CREDENTIALS --fail -o /dev/null -X PUT "$RELEASES_HOST/releases/$product@$VERSION/assets/$(basename $file)"
+        curl $CREDENTIALS --fail -o /dev/null -X PUT "$RELEASES_HOST/releases/$product@$VERSION/assets/$(basename "$file" | sed 's/ /%20/g')"
```
Comment (tcsc) [inline .drone.yml:1234]: Is there a more general way to urlencode the filename?

File: .drone.yml [hunk 3]
```diff
@@ -1206,20 +1206,23 @@ steps:
   - which curl || apk add --no-cache curl
   - |-
     cd "$WORKSPACE_DIR/go/artifacts"
-    for file in $(find . -type f ! -iname '*.sha256' ! -iname '*-unsigned.zip*'); do
+    find . -type f ! -iname '*.sha256' ! -iname '*-unsigned.zip*' | while read -r file; do
```
Comment (tcsc) [inline .drone.yml:1209]: What's the difference between `for file in ...` vs `find ... | while read ...` - asking for a `bash` deficient person  (i.e. me)

**Review by tcsc — APPROVED** [review]

File: .drone.yml (hunk 3 above)
Comment (justinas) [reply .drone.yml:1209]: Not an expert either, but if I remember correctly: `for` will split by `IFS`, which by default includes spaces. One could in theory use something like `IFS=$'\n' for ...` to only split elements by newline, but I think this did not work as easily with alpine's shell (which does not have some of bash's stuff).

`read X Y Z` will still split the line by word, but if you have more output variables than words, the remainder will end up in the last variable. E.g. with input `foo bar baz quux`, `read X Y Z` will end up in `Z="baz quux"`. Thus, with only one output variable (`read X`), no splitting is performed, and in effect the whole line ends up in `X` unsplitted.

File: .drone.yml (hunk 2 above)
Comment (justinas) [reply .drone.yml:1234]: I couldn't find an easily available standalone utility to urlencode, aside from snippets like this https://stackoverflow.com/a/10660730/1264974

IMO it's fine for now, as our filenames are pretty uniform, and `Teleport Connect` is the first thing that needs to be urlencoded (all others roughly fall into the `[A-z0-9._-]` set)

**Review by LKozlowski — APPROVED** [review]

**Review by espadolini — APPROVED** [review]

**Pull request merged**